scripts/migration/lib.sh

```diff
@@ -281,6 +281,43 @@ check_version() {
     fi
   fi
 
+  # Make sure **config** and **cluster** is on the same (old) version
+  cluster_version="$(get_apps_version "${1}" >/dev/null 2>&1 || true)"
+  if [[ "${2}" == "prepare" ]]; then
```

Comment:
If we are in an apply do we not want to do some validation as well here?

Reply:
What validation would we run here for apply?

Reply:
What validation is done up to this part in the apply case?

Reply:
I have reviewed the checks and I think they are fine, I have one suggestion on moving the ensure prepared to gather them in the same version check function, and to make it fail faster should it not be prepared.